pkg/grpc/grpc.go

```diff
@@ -108,31 +109,21 @@ func (g *Manager) GetAppChannel() (channel.AppChannel, error) {
 // GetAppClient returns the gRPC connection to the local app.
 // If there's no active connection to the app, it creates one.
 func (g *Manager) GetAppClient() (grpc.ClientConnInterface, error) {
-	if g.localConnCreateFn != nil {
-		return g.localConn.Get(g.localConnCreateFn)
-	}
-	return g.localConn.Get(g.defaultLocalConnCreateFn)
-}
+	if g.appClientConn == nil {
+		c, err := g.defaultLocalConnCreateFn()
```

Comment:
Its this line that cannot be invoked concurrently, as it's called once on init and any subsequent calls will not satisfy the nil check. @ItalyPaleAle

Reply:
But if 2 goroutines call `GetAppClient` at the exact same time, `g.appClient` will be nil for both, so both will get here and both will try to write  `g.appClientConn = c`?

Reply:
When these goroutines get there, it will not be nil. Before any code that can reach this line is executed, it would already have been initialized during init in runtime

Reply:
Ok, I see